core/src/main/java/io/bitcoinsv/headerSV/core/service/HeadersSvServiceImpl.java

```diff
@@ -224,17 +224,19 @@ private void runAndRetry(String workName, int maxAttempts, int millisWaitBetween
             try {
                 numAttempts++;
                 workToDo.run();
+                workDoneOK = true;
             } catch (Exception e) {
                 if (numAttempts > maxAttempts) { throw e; }
                 log.warn("ERROR while {}, retrying... ({} attempts left)", workName, (maxAttempts - numAttempts));
                 // a bit of a delay between runs:
-                try { Thread.sleep(millisWaitBetweenRuns);} catch (InterruptedException ie) {}
+                try { Thread.sleep(millisWaitBetweenRuns); } catch (InterruptedException ie) {}
             }
         }
     }
 
     // It process an incoming HeadersMsg:
     private void processHeadersMsg(HeadersMsg headerMsg, PeerAddress peerAddress) {
+        log.debug("Processing headers msg from peer {}: {}", peerAddress.toString(), headerMsg.getBlockHeaderMsgList().stream().findFirst().get().getHash());
```

Comment:
Are we sure that headers msg will always contain some block header msg? Otherwise, get() will throw an exception.